src/UrlRpcEncoder.ts

```diff
@@ -83,18 +83,20 @@ export class UrlRpcEncoder {
     }
 
     public static prepareRedirectReply(state: State, status: ResponseStatus, result: any): string {
-        const params = new URLSearchParams();
+        const returnUrl = new URL(state.returnURL!);
+        const params = returnUrl.searchParams;
         params.set('status', status);
         params.set('result', JSONUtils.stringify(result));
         params.set('id', state.id.toString());
 
-        return `${state.returnURL!}${new URL(state.returnURL!).search.length > 0 ? '&' : '?'}${params.toString()}`;
+        return `${returnUrl.origin}${returnUrl.pathname}?${params.toString()}`;
```

Comment:
why not simply return returnUr.href?

Reply:
`href` includes the query string (search params). But when retrieving a redirect-result from RPC, the receiver (client) reads the params in a ignore-later-doubled-params method. Thus it is not enough to just append the current result to the `href` and thus the potential query-string, as that string might already contain the params that we are now writing, thus loosing the new result.

Thus we need to "reconstruct" the returnURL, and since there is no way to get the URL without the search-query from the URL object, we need to do it in two steps and then append the new query-string.